src/Mutations/CreateTokenMutationCreator.php

```diff
@@ -1,83 +1,137 @@
+     * @param mixed       $object
+     * @param array       $args
+     * @param mixed       $context
      * @param ResolveInfo $info
-     * @return null|Member|static
-     * @throws \Psr\Container\NotFoundExceptionInterface
+     * @return array
+     * @throws NotFoundExceptionInterface
+     * @throws ValidationException
      */
-    public function resolve($object, array $args, $context, ResolveInfo $info)
+    public function resolve($object, array $args, $context, ResolveInfo $info): array
     {
-        /** @var Security $security */
-        $security = Injector::inst()->get(Security::class);
-        $authenticators = $security->getApplicableAuthenticators(Authenticator::LOGIN);
+        // Authenticate this member
         $request = Controller::curr()->getRequest();
-        $member = null;
-
-        if (count($authenticators)) {
-            /** @var Authenticator $authenticator */
-            foreach ($authenticators as $authenticator) {
-                $member = $authenticator->authenticate($args, $request, $result);
-                if ($result->isValid()) {
-                    break;
-                }
-            }
+        $member = $this->getAuthenticatedMember($args, $request);
+
+        // Handle unauthenticated
+        if (!$member) {
+            return $this->generateResponse(TokenStatusEnum::STATUS_BAD_LOGIN);
         }
-        $authenticator = Injector::inst()->get(JWTAuthenticator::class);
-
-        if ($member instanceof Member) {
-            $member->Token = $authenticator->generateToken($member);
-        } elseif (JWTAuthenticator::config()->get('anonymous_allowed')) {
-            $member = Member::create(['ID' => 0, 'FirstName' => 'Anonymous']);
-            // Create an anonymous token
-            $member->Token = $authenticator->generateToken($member);
-        } else {
-            Security::setCurrentUser(null);
-            Injector::inst()->get(IdentityStore::class)->logOut();
-
-            // Return a token-less member
-            $member = Member::create();
+
+        // Create new token from this member
+        $authenticator = $this->getJWTAuthenticator();
+        $token = $authenticator->generateToken($request, $member);
+        return $this->generateResponse(TokenStatusEnum::STATUS_OK, $member, $token->__toString());
+    }
+
+    /**
+     * Get an authenticated member from the given request
+     *
+     * @param array       $args
+     * @param HTTPRequest $request
+     * @return Member|MemberExtension
+     */
+    protected function getAuthenticatedMember(array $args, HTTPRequest $request): ?Member
+    {
+        // Login with authenticators
+        foreach ($this->getLoginAuthenticators() as $authenticator) {
+            $result = new ValidationResult();
+            $member = $authenticator->authenticate($args, $request, $result);
+            if ($member && $result->isValid()) {
+                return $member;
+            }
         }
 
```

Comment:
Would it be cleaner to return the result no matter what? setting an error that no valid member has been found